### PR TITLE
Spring Boot Data JPA

### PR DESCRIPTION
**Pull request by ponomvrenko**
- Replaced BookRepositoryImpl with JpaRepository interface;
- Implemented liquibase dependencies & configurations (changelog-master, changes);
- Optimized book entity fields by using @Column;
- Added PUT & DELETE (soft) methods in controller layer;

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -38,4 +38,16 @@ public List<BookDto> findAll() {
...
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setId(id);
```
Comment (DPotapov93) [inline src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java:50]: Maybe we need an existing ID in the database check, but I am not sure

**Review by DPotapov93 — APPROVED** [review]
Well done! 👍

File: src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java (hunk 1 above)
Comment (vovasalyha) [reply src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java:50]: Yes, we need it. (DB will throw an exception anyway, but by checking it here we can throw an exception with valuable message)

**Review by vovasalyha — APPROVED** [review]
Approving, but take a look on my comment regarding update method.

**Pull request merged**